src/QMCWaveFunctions/Fermion/DiracDeterminantBatched.cpp

```diff
@@ -47,15 +52,23 @@ void DiracDeterminantBatched<DET_ENGINE>::set(int first, int nel, int delay)
 }
 
 template<typename DET_ENGINE>
-void DiracDeterminantBatched<DET_ENGINE>::invertPsiM(const Matrix<Value>& logdetT)
+void DiracDeterminantBatched<DET_ENGINE>::invertPsiM(const DualMatrix<Value>& psiM, DualMatrix<Value>& psiMinv)
 {
   ScopedTimer inverse_timer(InverseTimer);
-  det_engine_.invert_transpose(logdetT, log_value_);
+  det_engine_.invert_transpose(psiM, psiMinv, log_value());
+
+#ifndef NDEBUG
+  // This is easily breakable in that it assumes this function gets psiMinv == det_engine_.psiMinv_
+  auto& engine_psiMinv = det_engine_.psiMinv();
+  dummy_vmt.attachReference(engine_psiMinv.data(), engine_psiMinv.rows(), engine_psiMinv.cols());
```

Comment:
Say you have a DiracDeterminantBase reference or pointer, as I have had numerous times in debug/testing you want to be able to inspect the psiMinv in a generic way even though where and how that is stored is implementation specific. calling the virtual method (debug only) getPsiMInv() gets you a reference to a CPU psiMInv.
The call in DiracDeterminantBase is getPsiMInv() instead of get_psiMinv to indicate that this isn't a practically zero cost accessor. I had many unit tests at one point that were more thorough comparing determinant implementations when in debug.  They're lost back who knows how many rejected PR's ago.  This is still useful.  But I go out of my way to make it unusable for anything but debug testing and debugging.

Reply:
I support removing this whole apparatus once the inversion is in and stable.

Reply:
defer to #3461